### ISSUE TITLE
Does any parameters on strandedness?

### ISSUE THREAD

**Issue by WangKang-Leo** [liulab-dfci/TRUST4#192]
Hello, I have run the TRUST4 based on our bulk RNAseq cohort (PE150bp, ribo-zero), where fastq is input, but the results are not consistent with other immune metrics (GEP based). I want to optimize the analyses by adding some parameters, such as strandedness. Our data have significant batch effect, but it should be fine for TRUST4. Or do you have any ideas?

Thanks in advance

**Comment by mourisl** [comment]
Could you please elaborate more on the issues? For example, for the inconsistency with GEP, do you mean the V gene or C gene expression from gene expression data is different from TRUST4's prediction? For strandedness, do you mean your RNA-seq is stranded RNA-seq?

**Comment by WangKang-Leo** [comment]
I did not look at V/C gene, but using some typical immune biomarkers (CD3,CD8, CD20) in mRNA or protein level, or GEP immune signature score for T cell, B cell. 

Our RNA-seq is reverseStranded, and I don't know whether it has influence or not on TRUST4.

**Comment by mourisl** [comment]
The strand-specific RNA-seq probably does not affect TRUST4 much, it is rare to have a read whose reverse-complement matches better to VDJ genes than its true orientation. 

For the GEP analysis, do you mean when other data show no or few T or B cells, but TRUST4 still assembles many TCRs and BCRs?

**Comment by WangKang-Leo** [comment]
Yes, not well correlated between GEP-based T/B cell and those from TRUST4.....

**Comment by mourisl** [comment]
How did you calculate the abundance from TRUST4? Could you please show me a few high abundant TRBs and IGHs from the airr output? I can check to see whether those are misassemblies. Thank you.

**Comment by WangKang-Leo** [comment]
I used code from RIMA (https://github.com/liulab-dfci/RIMA_pipeline) as following: 
![1681928636939](https://user-images.githubusercontent.com/49832670/233166203-80f55b74-8ed2-4aa8-8f86-0be1df47f3e5.png)
Also, I attached the sample with highest T cell infiltration (0.000025). The cohort includes 200 samples.
[UF-3016-1226-0_airr_align.txt](https://github.com/liulab-dfci/TRUST4/files/11276495/UF-3016-1226-0_airr_align.txt)

**Comment by WangKang-Leo** [comment]
[UF-3016-1226-0_airr.txt](https://github.com/liulab-dfci/TRUST4/files/11276565/UF-3016-1226-0_airr.txt)
Thanks so much, here you are!

**Comment by WangKang-Leo** [comment]
Hello again, any news or findings from my AIRR output?

**Comment by mourisl** [comment]
Those assemblies look authentic to me. I guess this sample is sequenced pretty deep, so even though the T or B cell infiltration level is low, it still captures a lot. Also, the abundance is affected by receptor gene expression. For example, the top IGH/IGK CDR3s may come from plasma B cells, which may have super high expression but the actual cell number could be much fewer. Hope this helps.

**Comment by WangKang-Leo** [comment]
Thanks for your prompt feedback. Do you have any suggestions to adjust or filter the AIRR results?

**Comment by mourisl** [comment]
I usually filter the non-productive chains in the results. For BCRs, you may cluster the similar CDR3s as they might be come from the same lineage during somatic hypermutation (which is another definition of clonotype), but this only changes the number of distinct clonotypes and won't affect the total abundances.
